source/extensions/filters/http/local_ratelimit/local_ratelimit.cc

```diff
@@ -73,20 +71,37 @@ FilterConfig::FilterConfig(
     throw EnvoyException("local rate limit token bucket must be set for per filter configs");
   }
 
+  absl::Status creation_status;
+  rate_limit_config_ = std::make_unique<Filters::Common::RateLimit::RateLimitConfig>(
+      config.rate_limits(), context, creation_status);
+  THROW_IF_NOT_OK_REF(creation_status);
+
+  if (rate_limit_config_->empty()) {
+    if (!config.descriptors().empty()) {
+      ENVOY_LOG_FIRST_N(
+          warn, 20,
+          "'descriptors' is set but no valid 'rate_limits' in LocalRateLimit config. Note that "
+          "'descriptors' only makes sense when 'rate_limits' in LocalRateLimit config or route "
+          "config is specified. And please take the 'rate_limits' in the LocalRateLimit config as "
+          "priority because the 'rate_limits' in the route config may be ignored by the filter in "
+          "the future.");
+    }
+  }
```

Comment:
Sorry for being dense but I still don't understand why this has to be a runtime failure? This is a new feature, right? If the user configured it incorrectly can we fail to load the config?

/wait-any

Reply:
For backward compatibility. 

Note, in the previous local ratelimit, the local ratelimit's `descriptors` field will be used to declare the rate limit config of specific descriptor, like 10 requests per second for descriptor (key1, value1), etc.

And the `rate_limits` in the route configuration will be used to generate specific descriptor (key, value) of request and the descriptor will be used to match a config in the `descriptors` field (The naming is a little confusing because the historical reason.)

But previous implementation has some drawbacks in actual practices (see issue description), so I added the new `rate_limits` field in the filter config directly  as an alternative and resolve previous drawbacks.

But we cannot reject the filter config which doesn't contains the new `rate_limits`, because the users may have configured `rate_limits` in the route.